src/bin/pg_autoctl/monitor.c

```diff
@@ -2998,9 +2998,6 @@ monitor_get_extension_version(Monitor *monitor, MonitorExtensionVersion *version
 		return false;
 	}
 
-	/* disconnect from PostgreSQL now */
-	pgsql_finish(&monitor->pgsql);
```

Comment:
This removal also has effect on `monitor_ensure_extension_version`, because this function is called there twice too.

Reply:
The `monitor_ensure_extension_version` in turn is only called in `monitor_service_run` which implements a LISTEN/NOTIFY client loop, where we keep the same connection for a long time anyway. I think it's okay even in that case that we're not closing the monitor connection anymore in the lower-level functions, the code would just re-open the connection at the next step anyway...